src/sds_data_model/_raster.py

```diff
@@ -0,0 +1,466 @@
+from pathlib import Path
+from typing import Dict, List, Literal, Optional, Union
+
+from affine import Affine
+from cv2 import INTER_LINEAR, INTER_NEAREST, resize
+from numpy import array, full
+from numpy.typing import NDArray
+from rasterio.drivers import raster_driver_extensions
+from rioxarray.rioxarray import affine_to_coords
+from xarray import DataArray, Dataset, merge, open_dataset
+
+from sds_data_model._vector import _check_layer_projection
+from sds_data_model.constants import BNG_XMAX, BNG_XMIN, BNG_YMAX, CELL_SIZE
+
+
+def _has_wrong_cell_size(
+    transform: Affine,
+    expected_cell_size: int,
+) -> bool:
+    """Return true if the cell size of x (a) and y (e) in the \
+    input transform are **not equal** to the expected cell size.
```

Comment:
This should be a one line (<88 character) description of the function and you should need to use `\` to break a line in a multiline docstring. You can add more information in the optional description paragraph.

```suggestion
    """Return `True` if the cell size is not what is expected.
    
    Return `False` if either the x (a) and y (e) values of the input transform do
    **not equal** the expected cell size.
```